File: modules/server/eventdata/weapon_range.lua
WeaponRange = {}

if not ServerConfig.Modules.WeaponRange.enabled then
    return
end

local tazer = 911657153

function WeaponRange.ProcessEventData(sender, data)
    local rawTargetData = data["hitGlobalId"] or data["hitGlobalIds"][1]
    local ped, target = GetPlayerPed(sender), NetworkGetEntityFromNetworkId(rawTargetData)

    if DoesEntityExist(target) and IsPedAPlayer(target) then
        local pCoords = GetEntityCoords(ped)
        local tCoords = GetEntityCoords(target)
        local dist = Util.GetDistance(pCoords.x, pCoords.y, tCoords.x, tCoords.y)

        if dist > 400.0 then
            TriggerEvent("icarus:my602oxd71pv", sender, "Weapon Range [C1]", false, {
                range = dist
             })
            CancelEvent()
        end
        if data["weaponType"] == tazer and dist > ServerConfig.Modules.WeaponRange.maxTazerRange then
            TriggerEvent("icarus:my602oxd71pv", sender, "Weapon Range [C2]", false, {
                tazedRange = dist,
                maxRange = ServerConfig.Modules.WeaponRange.maxTazerRange
             })
            CancelEvent()
        end
    end
end

File: modules/server/eventdata/weapon_range.js
const tazerHash = 911657153;

module.exports = (sender, data) => {
    const weaponHash = data.weaponType;
    const netId = data.hitGlobalId || data.hitGlobalIds[0];
    const target = NetworkGetEntityFromNetworkId(netId);

    if (!DoesEntityExist(target) || !IsPedAPlayer(target)) { return; }

    const ped = GetPlayerPed(sender);
    const pCoords = GetEntityCoords(ped);
    const tCoords = GetEntityCoords(target);
    const distance = Math.sqrt((pCoords[0] - tCoords[0]) ** 2 + (pCoords[1] - tCoords[1]) ** 2);

    if (distance > 400) {
        emit("icarus:my602oxd71pv", sender, "Weapon Range [C1]", false, {
            distance: distance
        });
        CancelEvent();
    } else if (weaponHash == tazerHash && distance > serverConfig.Modules.WeaponRange.maxTazerRange) {
        emit("icarus:my602oxd71pv", sender, "Weapon Range [C2]", false, {
            distance: distance,
            tazer: serverConfig.Modules.WeaponRange.maxTazerRange
        });
        CancelEvent();
    }
};